accepted/exception-handling.md

```diff
@@ -0,0 +1,514 @@
+observations:
+
+* For the foreseeable moment, we do not plan to remove or replace
+  Wasmtime's frame-pointer-based stack walking as part of implementing
+  the Wasm exceptions proposal; unwinding exceptions will use frame
+  pointers to walk the stack, the same way that existing stack walking
+  happens today.[^fp-unwind-future]
+
+* Again, at least for the foreseeable moment, we can make
+  `try_call[_indirect]` instructions use a variant of the target's
+  Wasm calling convention where all registers are caller-saved.
+
+When taken together, these observations mean that our exception side
+tables *only* need to encode a map from a `try_call`'s return address
+PC to the set of exception tags that the `try_call` is catching and
+the PC of the associated landing pad. The side tables do *not* need to
+encode information about how to find the previous frame from the
+current one (what DWARF calls the "canonical frame address") because
+we will just use frame pointers for that. And the side tables do *not*
+need to provide a mechanism for restoring callee-save registers before
+control is transferred to the landing pad because there are no
+callee-save registers.
+
+Using a DWARF subset's advantages are that Wasmtime developers would
+be able to reuse tools like `dwarfdump` for inspecting the side tables
+in `.cwasm` files. Its downsides are that it just has way more stuff
+than we need, not all of which we can just ignore. To encode the
+which-exceptions-this-location-catches information, we would have to
+emit Frame Description Entries (FDEs) with our custom personality
+routines. But FDEs have non-optional fields, such as a list of CFI
+instructions, that are used for recovering the previous frame and its
+registers, which we don't need because we are always doing
+frame-pointer walking. Beyond impedance mismatch cognitive overhead,
+this leads to larger side tables and bloated `.cwasm` binaries.
+
+**Therefore, in the interests of simplicity, faster lookups, and slim
+`.cwasm` binaries, we will design our own bespoke format for the side
+tables.** It is worth noting that we essentially do this already for
+our existing traps metadata tables. We can ensure that the format is
+suitable for usage straight from `mmap`, without any need to translate
+the on-disk encoding into some in-memory data structure, which would
+slow down `Module` creation. We can make it contain only and exactly
+the information we actually need. We can use a format that is fast and
+easy to decode. All this results in smaller tables, faster lookups,
+and slimmer `.cwasm` binaries. While we won't be able to piggyback on
+tools like `dwarfdump` &mdash; given how little information we need to
+store in the side tables and the relative simplicity that brings, and
+the fact that we can always use `std::fmt::Debug` in tactically-placed
+logs &mdash; the lack of `dwarfdump` isn't the end of the
+world.[^dwarf-future]
```

Comment:
To comment on this specifically, we talked about this in the Cranelift meeting today and I filed https://github.com/bytecodealliance/wasmtime/issues/10336 which is certainly extra work but I think is "The Answer" for how to recover something that's `dwarfdump`-like with our own custom format